cmd/syft/cli/commands/packages.go

```diff
@@ -0,0 +1,254 @@
+    {{.appName}} {{.command}} docker-archive:path/to/yourimage.tar     use a tarball from disk for archives created from "docker save"
+    {{.appName}} {{.command}} oci-archive:path/to/yourimage.tar        use a tarball from disk for OCI archives (from Skopeo or otherwise)
+    {{.appName}} {{.command}} oci-dir:path/to/yourimage                read directly from a path on disk for OCI layout directories (from Skopeo or otherwise)
+    {{.appName}} {{.command}} singularity:path/to/yourimage.sif        read directly from a Singularity Image Format (SIF) container on disk
+`
+	nonImageSchemeHelp = `    {{.appName}} {{.command}} dir:path/to/yourproject                  read directly from a path on disk (any directory)
+    {{.appName}} {{.command}} file:path/to/yourproject/file            read directly from a path on disk (any single file)
+`
+	packagesSchemeHelp = "\n  " + schemeHelpHeader + "\n" + imageSchemeHelp + nonImageSchemeHelp
+
+	packagesHelp = packagesExample + packagesSchemeHelp
+)
+
+type packagesOptions struct {
+	options.Output      `yaml:",inline" mapstructure:",squash"`
+	options.UpdateCheck `yaml:",inline" mapstructure:",squash"`
+	options.Packages    `yaml:",inline" mapstructure:",squash"`
+}
+
+func packagesOptionsDefault() *packagesOptions {
+	return &packagesOptions{
+		Output:      options.OutputDefault(),
+		UpdateCheck: options.UpdateCheckDefault(),
+		Packages:    options.PackagesDefault(),
+	}
+}
+
+//nolint:dupl
+func Packages(app clio.Application) *cobra.Command {
+	opts := packagesOptionsDefault()
+
+	return app.SetupCommand(&cobra.Command{
+		Use:   "packages [SOURCE]",
+		Short: "Generate a package SBOM",
+		Long:  "Generate a packaged-based Software Bill Of Materials (SBOM) from container images and filesystems",
+		Example: internal.Tprintf(packagesHelp, map[string]interface{}{
+			"appName": app.ID().Name,
+			"command": "packages",
+		}),
+		Args: validatePackagesArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			applicationUpdateCheck(app, opts.CheckForAppUpdate)
+
+			return runPackages(app, opts, args[0])
+		},
+	}, opts)
+}
+
+func validatePackagesArgs(cmd *cobra.Command, args []string) error {
+	return validateArgs(cmd, args, "an image/directory argument is required")
+}
+
+func validateArgs(cmd *cobra.Command, args []string, error string) error {
+	if len(args) == 0 {
+		// in the case that no arguments are given we want to show the help text and return with a non-0 return code.
+		if err := cmd.Help(); err != nil {
+			return fmt.Errorf("unable to display help: %w", err)
+		}
+		return fmt.Errorf(error)
+	}
+
+	return cobra.MaximumNArgs(1)(cmd, args)
+}
+
+// nolint:funlen
+func runPackages(app clio.Application, opts *packagesOptions, userInput string) error {
+	err := validatePackageOutputOptions(&opts.Output)
+	if err != nil {
+		return err
+	}
+
+	writer, err := options.MakeSBOMWriter(opts.Outputs, opts.File, opts.OutputTemplatePath)
```

Comment:
I know that `options/writer.go` didn't change, but should we consider make these expressible via only option structs? E.g. `options/output.go` on the `Output` struct add a new `SBOMWriters()` method that calls `MakeSBOMWriter` and returns a single writer? This way we could unexport `MakeSBOMWriter` entirely.

Reply:
I made a change for this. I think it's better, but it's also a tiny bit wonky because of how the `MakeSBOMWriter` was being called differently across multiple commands.

Reply:
agreed, I like this better 👍